File: src/data/hullmods/niko_MPC_graviticSupercomputer.java
package data.hullmods;

import com.fs.starfarer.api.combat.BaseHullMod;
import com.fs.starfarer.api.combat.MutableShipStatsAPI;
import com.fs.starfarer.api.combat.ShipAPI;

public class niko_MPC_graviticSupercomputer extends BaseHullMod {

    public final float rangeMult = 6.5f;
    public final float pdRangeMult = 0.7f;
    public final float visionIncrement = 5000f;
    public final float recoilMult = 0.3f;

    @Override
    public void applyEffectsBeforeShipCreation(ShipAPI.HullSize hullSize, MutableShipStatsAPI stats, String id) {
        stats.getBallisticWeaponRangeBonus().modifyMult(id, rangeMult);
        stats.getEnergyWeaponRangeBonus().modifyMult(id, rangeMult);
        stats.getMissileWeaponRangeBonus().modifyMult(id, rangeMult);

        stats.getNonBeamPDWeaponRangeBonus().modifyMult(id, pdRangeMult);

        stats.getSightRadiusMod().modifyFlat(id, visionIncrement);

        stats.getRecoilPerShotMult().modifyMult(id, recoilMult);
    }

    public String getDescriptionParam(int index, ShipAPI.HullSize hullSize) {
        if (index == 0) return "" + (int)Math.round(rangeMult) + "00" + "%";
        else if (index == 1) return "" + (int)Math.round(recoilMult) + "00" + "%";
        return null;
    }
}

File: src/data/hullmods/niko_MPC_graviticSupercomputer.kt
package data.hullmods

import com.fs.starfarer.api.combat.BaseHullMod
import com.fs.starfarer.api.combat.MutableShipStatsAPI
import com.fs.starfarer.api.combat.ShipAPI
import com.fs.starfarer.api.combat.ShipAPI.HullSize
import com.fs.starfarer.api.combat.WeaponAPI
import com.fs.starfarer.api.combat.WeaponAPI.*

class niko_MPC_graviticSupercomputer : BaseHullMod() {
    val rangeMult = 3f
    val pdRangeMult = 1f
    val visionIncrement = 10000f
    val recoilMult = 0.3f
    val autofireMult = 1f

    val maxRecoilMult = 0.05f
    override fun applyEffectsBeforeShipCreation(hullSize: HullSize, stats: MutableShipStatsAPI, id: String) {
        stats.ballisticWeaponRangeBonus.modifyMult(id, rangeMult)
        stats.energyWeaponRangeBonus.modifyMult(id, rangeMult)
        stats.missileWeaponRangeBonus.modifyMult(id, (rangeMult))
        stats.nonBeamPDWeaponRangeBonus.modifyMult(id, pdRangeMult)
        stats.sightRadiusMod.modifyFlat(id, visionIncrement)
        stats.recoilPerShotMult.modifyMult(id, recoilMult)
        stats.recoilDecayMult.modifyMult(id, recoilMult)
        stats.maxRecoilMult.modifyMult(id, maxRecoilMult)
        stats.autofireAimAccuracy.modifyFlat(id, autofireMult)
    }

    override fun getDescriptionParam(index: Int, hullSize: HullSize): String? {
        if (index == 0) return "" + Math.round(rangeMult) + "00" + "%" else if (index == 1) return "" + Math.round(
            recoilMult
        ) + "00" + "%"
        return null
    }
}